http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/NonLeoMessageSubscriber.scala

```diff
@@ -0,0 +1,144 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package monitor
+
+import cats.effect.{Concurrent, Timer}
+import cats.implicits._
+import cats.mtl.Ask
+import fs2.Stream
+import io.chrisdavenport.log4cats.StructuredLogger
+import io.circe.Decoder
+import org.broadinstitute.dsde.workbench.google2.{Event, GoogleComputeService, GoogleSubscriber, InstanceName, ZoneName}
+import org.broadinstitute.dsde.workbench.leonardo.JsonCodec._
+import org.broadinstitute.dsde.workbench.leonardo.db.{clusterQuery, kubernetesClusterQuery, DbReference}
+import org.broadinstitute.dsde.workbench.leonardo.http.dbioToIO
+import org.broadinstitute.dsde.workbench.leonardo.monitor.NonLeoMessage.{CryptoMining, DeleteKubernetesClusterMessage}
+import org.broadinstitute.dsde.workbench.leonardo.util.{DeleteClusterParams, GKEAlgebra}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+/**
+ * This is subscriber for messages that are not published by Leonardo itself.
+ * But rather from cron jobs, or cloud logging sink (triggered from crypto-mining activity)
+ */
+class NonLeoMessageSubscriber[F[_]: Timer](gkeAlg: GKEAlgebra[F],
+                                           computeService: GoogleComputeService[F],
+                                           subscriber: GoogleSubscriber[F, NonLeoMessage])(
+  implicit logger: StructuredLogger[F],
+  F: Concurrent[F],
+  metrics: OpenTelemetryMetrics[F],
+  dbRef: DbReference[F]
+) {
+  val process: Stream[F, Unit] = subscriber.messages
+    .parEvalMapUnordered(10)(messageHandler)
+    .handleErrorWith(error => Stream.eval(logger.error(error)("Failed to initialize message processor")))
+
+  private[monitor] def messageHandler(event: Event[NonLeoMessage]): F[Unit] = {
+    val traceId = event.traceId.getOrElse(TraceId("None"))
+    for {
+      now <- nowInstant[F]
+      implicit0(ev: Ask[F, AppContext]) <- F.pure(Ask.const[F, AppContext](AppContext(traceId, now, None)))
+      _ <- metrics.incrementCounter(event.msg.messageType)
```

Comment:
Could we name it like `s"LeoPubSub/${event.msg.messageType}"`? Just to distinguish it when making dashboards.